### PR TITLE
Add pyeo

### PR DESCRIPTION
**Pull request by Thingus**
Pyeo is a library for downloading, preprocessing and classifying satellite images and other geospatial rasters. It presently includes  tools to:

  - Download Sentinel-2 and Planet rasters (requires a Planet API key)

  - Preprocess, stack, mosaic and composite rasters (does not have to be S2 or planet) 

 -  Build models for geospatial data collected from digitized maps

  - Classify pixels in raster with created models 

  - Can classify land use (with single images) or land change (with a pair of images) 

  - Produce validation points (with confidence) of generated rasters 

Pyeo is built on a stack of Numpy, Gdal and scikit-learn. It makes extensive use of virtual memory mappings via the gdal GetVirtualMemArray() function, so it not suitable for building on Windows. OSX is presently untested.

Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages: I'm not clear on what this means, but I don't think I've done it?
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

(Edited to include a brief description of the library; I didn't realise that this wasn't a totally automated process)

### CODE REVIEW

**Comment by Thingus** [comment]
Now this is passing CI and I know it's possible, I'm getting explicit permission from the dependency library owners (geomet, html2text, sentinelhub and sentinelsat) for their libraries to be made available on conda-forge - all are open-source with appropriate licensing, but it's good manners.

File: recipes/sentinelhub/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,73 @@
+{% set name = "sentinelhub" %}
+{% set version = "2.5.3" %}
```
Comment (AleksMat) [inline recipes/sentinelhub/meta.yaml]: A few days ago we released a new version 2.6.0 [releases](https://github.com/sentinel-hub/sentinelhub-py/releases). The SHA from PyPI for the new version is `94b1ff8402291683a801ae6293a3982360b9b7d90657293c04e16edb55a7ed10`

File: recipes/sentinelhub/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    - sentinelhub.config --help
+    - sentinelhub.download --help
+
+about:
+  home: "https://github.com/sentinel-hub/sentinelhub-py"
+  license: "MIT"
+  license_family: "MIT"
+  license_file: "LICENSE.md"
+  summary: "Sentinel Hub Utilities: skeleton by Thingus"
+  doc_url: ""
```
Comment (AleksMat) [inline recipes/sentinelhub/meta.yaml]: - doc_url is https://sentinelhub-py.readthedocs.io/en/latest/
- dev_url is https://github.com/sentinel-hub/sentinelhub-py

File: recipes/sentinelhub/meta.yaml (hunk 1 above)
Comment (Thingus) [reply recipes/sentinelhub/meta.yaml]: Addressed; thanks!

File: recipes/sentinelhub/meta.yaml (hunk 2 above)
Comment (Thingus) [reply recipes/sentinelhub/meta.yaml]: Addressed; thanks!

**Comment by iovsn** [comment]
Thanks for adding me as a maintainer.

**Comment by AleksMat** [comment]
I confirm to be added as a maintainer for sentinelhub recipe.

**Comment by andrejburja** [comment]
i would like to be added as a maintainer for this recipe.

**Comment by andrejburja** [comment]
Why there is no  `noarch: python`?

**Comment by Thingus** [comment]
The library I'm building makes heavy use of the gdal function GetVirtualMemArray(); this is presently not implemented on Windows, so noarch doesn't apply.
________________________________

**Comment by Thingus** [comment]
@conda-forge/staged-recipes I think this is ready for merging now, but I'm still unsure about the 'does not vend other packages' requirement; can you advise?

**Comment by Thingus** [comment]
@conda-forge-manager This recipie is now ready for merging.

File: recipes/geomet/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+test:
+  imports:
+    - geomet
+  commands:
+    - geomet --help
+
+about:
+  home: "https://github.com/geomet/geomet"
+  license: "Apache Software"
```
Comment (chrisburr) [inline recipes/geomet/meta.yaml]: Can you use the SPDX identifiers:
```suggestion
  license: Apache-2.0
```
(I assume it is 2.0)

File: recipes/geomet/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  license_family: "APACHE"
+  license_file: "PKG-INFO"
+  summary: "GeoJSON <-> WKT/WKB conversion utilities. Skeleton by Thingus as a dependency for Pyeo"
+  doc_url: ""
```
Comment (chrisburr) [inline recipes/geomet/meta.yaml]: Empty fields should be removed:
```suggestion
```

File: recipes/geomet/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  dev_url: ""
```
Comment (chrisburr) [inline recipes/geomet/meta.yaml]:
```suggestion
```

File: recipes/geomet/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  imports:
+    - geomet
+  commands:
+    - geomet --help
+
+about:
+  home: "https://github.com/geomet/geomet"
+  license: "Apache Software"
+  license_family: "APACHE"
+  license_file: "PKG-INFO"
```
Comment (chrisburr) [inline recipes/geomet/meta.yaml]: This file doesn't contain a copy of the licence, you can include a copy in the recipe directory as it's missing from the tarball.

File: recipes/geomet/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+source:
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: "f5ea069d01afecaab0c66a1ca516e2716a64aeecfa65021d15bc3b1d9aaa35e4"
+
+build:
+  number: 0
+  entry_points:
+    - geomet=geomet.tool:cli
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv "
+  skip: true  # [py<32]
```
Comment (chrisburr) [inline recipes/geomet/meta.yaml]: This looks like it can be noarch?
```suggestion
  noarch: python
```

File: recipes/geomet/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+requirements:
+  host:
+    - click
+    - pip
+    - python
```
Comment (chrisburr) [inline recipes/geomet/meta.yaml]: If it's noarch, you can constrain the version with:
```suggestion
    - python >=3.2
```

File: recipes/geomet/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+    - six
+  run:
+    - click
+    - python
```
Comment (chrisburr) [inline recipes/geomet/meta.yaml]: Similarly:
```suggestion
    - python >=3.2
```

File: recipes/geomet/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+    - geomet=geomet.tool:cli
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv "
+  skip: true  # [py<32]
+
+requirements:
+  host:
+    - click
+    - pip
+    - python
+    - six
```
Comment (chrisburr) [inline recipes/geomet/meta.yaml]: Similarly:
```suggestion
```

File: recipes/geomet/meta.yaml [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+build:
+  number: 0
+  entry_points:
+    - geomet=geomet.tool:cli
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv "
+  skip: true  # [py<32]
+
+requirements:
+  host:
+    - click
```
Comment (chrisburr) [inline recipes/geomet/meta.yaml]: Do you need this, for most pure Python packages `python` and `pip` is enough. (`host` is only used to execute the setup.py script)
```suggestion
```

File: recipes/pyeo/meta.yaml [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+  sha256: c759301d6669e5e3a1ac182a9976436a2b1093ccd553ffaa2921901c115b55e3
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+  skip: true  # [win]
+  skip: true  # [py<36]
+
+requirements:
+  build:
```
Comment (chrisburr) [inline recipes/pyeo/meta.yaml]: You don't need these build requirements

File: recipes/pyeo/meta.yaml [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+    - python
+    - pip
+  host:
+    - python
+    - pip
+    - numpy 1.14.*
```
Comment (chrisburr) [inline recipes/pyeo/meta.yaml]: If you're not compiling against numpy you don't need this
```suggestion
```

File: recipes/pyeo/meta.yaml [hunk 14]
```diff
@@ -0,0 +1,69 @@
...
+    - gdal
+    - joblib
+    - matplotlib
+    - pip
+    - pytest
+    - python
+    - requests
+    - scikit-image
+    - setuptools
+    - {{ pin_compatible('numpy') }}
```
Comment (chrisburr) [inline recipes/pyeo/meta.yaml]: Similarly:
```suggestion
    - numpy
```

File: recipes/pyeo/meta.yaml [hunk 15]
```diff
@@ -0,0 +1,69 @@
...
+    - scipy
+    - python-fmask
+    - sentinelhub
+    - sentinelsat
+    - tenacity
+
+test:
+  imports:
+    - pyeo
+  requires:
```
Comment (chrisburr) [inline recipes/pyeo/meta.yaml]: This isn't needed as it's already in the run requirements

File: recipes/pyeo/meta.yaml [hunk 16]
```diff
@@ -0,0 +1,69 @@
...
+    - python
+  commands:
+    - python -c "import pyeo"
```
Comment (chrisburr) [inline recipes/pyeo/meta.yaml]: This test is the same as the import test above. If it shouldn't be anything different (like run a test suite) it should be removed.

File: recipes/pyeo/meta.yaml [hunk 17]
```diff
@@ -0,0 +1,69 @@
+{% set name = "pyeo" %}
+{% set version = "0.65" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/clcr/pyeo/archive/v0.65-alpha.tar.gz
```
Comment (chrisburr) [inline recipes/pyeo/meta.yaml]: This URL should contain the version to allow the update bot to work:
```suggestion
  url: https://github.com/clcr/pyeo/archive/v{{ version }}.tar.gz
```

File: recipes/pyeo/meta.yaml [hunk 18]
```diff
@@ -0,0 +1,69 @@
...
+  sha256: c759301d6669e5e3a1ac182a9976436a2b1093ccd553ffaa2921901c115b55e3
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+  skip: true  # [win]
```
Comment (chrisburr) [inline recipes/pyeo/meta.yaml]: If the package can be made noarch, this line can be removed even if Windows then fails due to missing dependencies.

**Pull request closed without merging**